xbmc/utils/FileExtensionProvider.cpp

```diff
@@ -92,6 +94,57 @@ std::string CFileExtensionProvider::GetFileFolderExtensions() const
   return extensions;
 }
 
+bool CFileExtensionProvider::CanOperateExtension(const std::string& path) const
+{
+  /*!
+   * @todo Improve this function to support all cases and not only audio decoder
+   * with tracks inside.
+   */
+
+  // Get file extensions to find addon related to it.
+  std::string strExtension = URIUtils::GetExtension(path);
+  StringUtils::ToLower(strExtension);
+  if (!strExtension.empty() && CServiceBroker::IsBinaryAddonCacheUp())
+  {
+    std::vector<AddonInfoPtr> addonInfos;
+    m_addonManager.GetAddonInfos(addonInfos, true, ADDON_AUDIODECODER);
+    for (const auto& addonInfo : addonInfos)
+    {
+      if (CAudioDecoder::HasTracks(addonInfo))
+      {
+        const auto exts = StringUtils::Split(CAudioDecoder::GetExtensions(addonInfo), "|");
+        if (std::find(exts.begin(), exts.end(), strExtension) != exts.end())
+        {
+          /* Call addon to start a dir read about given file, if success, return
+           * as true.
+           */
+          CAudioDecoder result(addonInfo);
+          if (result.CreateDecoder() && result.ContainsFiles(CURL(path)))
+            return true;
+
+          /* If extension was supported and addon creation failed, we expect
+           * as not usable file and give as false here.
+           */
+          return false;
+        }
+      }
+    }
+
+    /*!
+     * About VFS addons we expect that they can and return true.
+     *
+     * @todo Check VFS addons can also be types in conflict with Kodi's
+     * supported parts!
```

Comment:
For clarity I would use these words

> We expect that VFS addons can support the file, and return true

Without the comma it reads as "We expect that VFS addons can support the file and we expect that VFS addons return true."

With the comma, it shifts the subject and reads as "We expect that VFS addons can support the file, and we return true."

Reply:
Got updated, thanks for the help.